### PR TITLE
[Feature] Support Flink job submission baseline implementation

### PR DESCRIPTION
**Pull request by yangguangfu007**
fix: https://github.com/apache/incubator-paimon-webui/issues/23

### Purpose

Submit flink Job base implementation

### CODE REVIEW

File: paimon-web-flink/paimon-web-flink-common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-table-planner_${scala.binary.version}</artifactId>
+            <version>${flink.version}</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/pom.xml:85]: New line at the end of file.

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+package org.apache.paimon.web.flink.context;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+/**
+ * The ApplicationExecutorContext class is an extension of the ExecutorContext, designed to work
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java:26]:
```suggestion
 * This application executor context implementation is an extension of {@link ExecutorContext}, which provides executor context
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * with Flink applications running in YARN or Kubernetes application mode.
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java:27]:
```suggestion
 * with Flink jobs running in application mode.
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * <p>In this mode, Flink automatically identifies the runtime environment and returns the
+ * appropriate StreamExecutionEnvironment instance.
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ApplicationExecutorContext.java:30]:
```suggestion
 * appropriated {@link StreamExecutionEnvironment} instance.
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ExecutorContext.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+package org.apache.paimon.web.flink.context;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+
+/** The ExecutorContext class provides the context for creating an Executor. */
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ExecutorContext.java:27]:
```suggestion
/** The context interface provides the context to create an Executor. */
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ExecutorContext.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    public StreamExecutionEnvironment getEnvironment() {
+        return environment;
+    }
+
+    public StreamTableEnvironment getTableEnvironment() {
+        return tableEnv;
+    }
+
+    private EnvironmentSettings createEnvironmentSettings(RuntimeExecutionMode mode) {
+        return mode == RuntimeExecutionMode.BATCH
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/ExecutorContext.java:59]: What's the return value of  `RuntimeExecutionMode.DYNAMIC`?

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/LocalExecutorContext.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.flink.context;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+/** The LocalExecutorContext class provides the context for creating a LocalExecutor. */
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/LocalExecutorContext.java:25]:
```suggestion
/** This local executor context implementation provides the context for creating a {@link LocalExecutor}. */
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/RemoteExecutorContext.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+package org.apache.paimon.web.flink.context;
+
+import org.apache.paimon.web.flink.context.params.RemoteParams;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+/** The RemoteExecutorContext class provides the context for creating a RemoteExecutor. */
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/RemoteExecutorContext.java:28]:
```suggestion
/** The remote executor context implementation provides the context for creating a {@link RemoteExecutor}. */
```

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/params/RemoteParams.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+package org.apache.paimon.web.flink.context.params;
+
+/** This class represents the parameters required to connect to a remote Flink cluster. */
+public class RemoteParams {
+
+    private final String host;
+    private final int port;
+    private String[] jarFilePath;
+
+    public RemoteParams(String host, int port, String[] jarFilePath) {
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/context/params/RemoteParams.java:28]: Why does the `jarFilePath` use string array?

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/executor/Executor.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+            for (String statement : statements) {
+                List<Operation> operations = parser.parse(statement);
+                if (operations.size() != 1) {
+                    throw new TableException("Single statement required.");
+                } else {
+                    Operation operation = operations.get(0);
+                    if (operation instanceof ModifyOperation) {
+                        modifyOperations.add((ModifyOperation) operation);
+                    } else {
+                        throw new TableException("INSERT statement required.");
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/executor/Executor.java:68]: Why does the message use `INSERT statement`?

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/executor/ExecutorFactory.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package org.apache.paimon.web.flink.executor;
+
+import org.apache.paimon.web.flink.context.ExecutorContext;
+
+/** The ExecutorFactory interface provides a method to create an Executor. */
+public interface ExecutorFactory<C extends ExecutorContext> {
+
+    Executor createExecutor(C context);
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/executor/ExecutorFactory.java:26]: Adds the comments of this interface.

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/operation/FlinkSqlOperationType.java [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+/**
+ * This enum represents the types of operations that can be performed in Flink SQL. It includes
+ * operations like SELECT, CREATE, DROP, ALTER, INSERT, etc. Each operation type is associated with
+ * a string value, and there are methods to compare the operation type with a string, check if the
+ * operation type is INSERT, and get the operation type from a SQL statement.
+ *
+ * <p>Typically, you would use this enum to categorize a SQL statement and perform different actions
+ * based on the operation type.
+ */
+public enum FlinkSqlOperationType {
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/operation/FlinkSqlOperationType.java:30]: Could this directly use the SqlOperation of Flink SQL?

File: paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/operation/SqlCategory.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.flink.operation;
+
+/** Enum representing the category of SQL statements. */
+public enum SqlCategory {
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/main/java/org/apache/paimon/web/flink/operation/SqlCategory.java:22]: Could the enum directly use the enum of Flink table?

File: paimon-web-flink/paimon-web-flink-common/src/test/java/org/apache/paimon/web/flink/parser/StatementParserTest.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** The test of {@link StatementParser}. */
+public class StatementParserTest extends TestBase {
+
+    @Test
+    public void testParse() {
+        String[] statements = StatementParser.parse(statement);
+        assertThat(statements.length).isEqualTo(5);
+        System.out.println(Arrays.stream(statements).collect(Collectors.toList()));
```
Comment (SteNicholas) [inline paimon-web-flink/paimon-web-flink-common/src/test/java/org/apache/paimon/web/flink/parser/StatementParserTest.java:37]: Could this assert the parsed content?

File: paimon-web-flink/pom.xml [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (SteNicholas) [inline paimon-web-flink/pom.xml:66]: New line at the end of file.

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@yangguangfu007, thanks for contribution. I left some comments for this pull request. PTAL.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
@yangguangfu007 Thanks for your contribution.

I am very worried about what it will bring if we maintain another SQL client like this, we can just use api to connect Flink sql-gateway, otherwise, a lot of work is required here to maintain these codes, research and development work, and bug fixing work.

File: paimon-web-flink/paimon-web-flink-common/pom.xml (hunk 1 above)
Comment (yangguangfu007) [reply paimon-web-flink/paimon-web-flink-common/pom.xml:85]: Thanks to the review, I will rewrite the code using sql-gateway.

**Comment by yangguangfu007** [comment]
Thanks to the review, I will rewrite the code using sql-gateway.

**Pull request closed without merging**